src/runmode-af-xdp.c

```diff
@@ -0,0 +1,424 @@
+        SCLogInfo("unable to find af-xdp config using default values");
+        goto finalize;
+    }
+
+    if_root = ConfFindDeviceConfig(af_xdp_node, iface);
+    if_default = ConfFindDeviceConfig(af_xdp_node, "default");
+
+    if (if_root == NULL && if_default == NULL) {
+        SCLogInfo("unable to find af-xdp config for "
+                  "interface \"%s\" or \"default\", using default values",
+                iface);
+        goto finalize;
+    }
+
+    /* If there is no setting for current interface use default one as main iface */
+    if (if_root == NULL) {
+        if_root = if_default;
+        if_default = NULL;
+    }
+
+    /* Threading */
+    int ret = ConfGetChildValueWithDefault(if_root, if_default, "threads", &confstr) != 1
+                      ? ConfigSetThreads(aconf, "auto")
+                      : ConfigSetThreads(aconf, confstr);
+
+    if (ret < 0) {
+        aconf->DerefFunc(aconf);
+        return NULL;
+    }
+    SC_ATOMIC_RESET(aconf->ref);
+    (void)SC_ATOMIC_ADD(aconf->ref, aconf->threads);
+
+    /* Promisc Mode */
+    (void)ConfGetChildValueBoolWithDefault(if_root, if_default, "disable-promisc", (int *)&boolval);
+    if (boolval) {
+        SCLogConfig("Disabling promiscuous mode on iface %s", aconf->iface);
+        aconf->promisc = 0;
+    }
+
+#ifdef HAVE_AF_XDP
+    /* AF_XDP socket mode options */
+    if (ConfGetChildValueWithDefault(if_root, if_default, "force-xdp-mode", &confstr) == 1) {
+        if (strncasecmp(confstr, "drv", 3) == 0) {
+            aconf->mode |= XDP_FLAGS_DRV_MODE;
+        } else if (strncasecmp(confstr, "skb", 3) == 0) {
+            aconf->mode |= XDP_FLAGS_SKB_MODE;
+        } else if (strncasecmp(confstr, "none", 4) == 0) {
+        } else {
+            SCLogWarning(SC_WARN_AFXDP_CONF,
+                    "Incorrect af-xdp xdp-mode setting, default (none) shall be applied");
+        }
+    }
+
+    /* copy and zerocopy binding options */
+    if (ConfGetChildValueWithDefault(if_root, if_default, "force-copy-mode", &confstr) == 1) {
+        if (strncasecmp(confstr, "zero", 4) == 0) {
+            aconf->bind_flags |= XDP_ZEROCOPY;
+        } else if (strncasecmp(confstr, "copy", 4) == 0) {
+            aconf->bind_flags |= XDP_COPY;
+        } else if (strncasecmp(confstr, "none", 4) == 0) {
+        } else {
+            SCLogWarning(SC_WARN_AFXDP_CONF,
+                    "Incorrect af-xdp copy-mode setting, default (none) shall be applied");
+        }
+    }
+
+    /* memory alignment mode selection */
+    if (ConfGetChildValueWithDefault(if_root, if_default, "mem-unaligned", &confstr) == 1) {
+        if (strncasecmp(confstr, "yes", 3) == 0) {
+            aconf->mem_alignment = XDP_UMEM_UNALIGNED_CHUNK_FLAG;
+        }
+    }
+
+    /* Busy polling options */
+    if (ConfGetChildValueBoolWithDefault(if_root, if_default, "enable-busy-poll", &conf_val) == 1) {
+        if (conf_val) {
+            aconf->enable_busy_poll = true;
```

Comment:
Should this one be true if unset and set to false if kernel < 5.11 ?

Reply:
This really depends on our desire with the feature. The user may not want the thread to hog the CPU (this will cause the thread to consume 100% of the CPU). I opted to provide the ability to enable/disable this feature, even if the kernel allows for it. This also applies to your comment above "Could we autodetect the feature by querying the kernel ?" - have you got a preference?

Reply:
I'm good with disabling capability. I think we should pick the default value that is bringing the more performance (and making it dependent of kernel availability).

Reply:
OK that works for me. I will provide the ability I currently have - the user can have it enabled/disabled, but also provide a method for automatic disabling if the kernel does not support the feature (< 5.11)